code/modules/xenoarchaeology/tools/xenoarchaeology_labeler.dm

```diff
@@ -0,0 +1,261 @@
+					if(!params["select"])
+						deselected_traits += trait_key
+						continue
+				else if(!(trait_key in deselected_traits))
+					if(params["select"])
+						selected_traits.Insert(1, trait_key)
+						label_traits.Insert(1, SSxenoarchaeology.xenoa_all_traits_keyed[trait_key])
+					else
+						deselected_traits += trait_key
+						continue
+				//Deselected traits
+				if(trait_key in deselected_traits)
+					deselected_traits -= trait_key
+		if("toggle_filter")
+			var/specific_filter = params["filter"]
+			if(specific_filter in enabled_trait_filters)
+				enabled_trait_filters -= specific_filter
+				filtered_traits += SSxenoarchaeology.labeler_traits_filter[params["filter"]]
+			else
+				enabled_trait_filters += specific_filter
+				filtered_traits -= SSxenoarchaeology.labeler_traits_filter[params["filter"]]
+	return TRUE
+
+/obj/item/xenoarchaeology_labeler/afterattack(atom/target, mob/user, proximity_flag)
+	. = ..()
+	if(proximity_flag && COOLDOWN_FINISHED(src, sticker_cooldown_timer))
+		COOLDOWN_START(src, sticker_cooldown_timer, 5 SECONDS)
+		create_label(target, user)
+	else if(!COOLDOWN_FINISHED(src, sticker_cooldown_timer))
+		to_chat(user, "<span class='warning'>The labeler is still printing.</span>")
+
+///reset all the options
+/obj/item/xenoarchaeology_labeler/proc/clear_selection()
+	label_traits = list()
+	selected_traits = list()
+	deselected_traits = list()
+	ui_update()
+
+/obj/item/xenoarchaeology_labeler/proc/create_label(mob/target, mob/user)
+	var/obj/item/sticker/xenoartifact_label/P = new(get_turf(src), label_traits)
+	if(target && user)
+		P.afterattack(target, user, TRUE)
+/*
+	Debug variant, spawns artifacts
+*/
+/obj/item/xenoarchaeology_labeler/debug
+	name = "xenoartifact debug labeler"
+	desc = "Use to create specific Xenoartifacts"
+	icon_state = "labeler_debug"
+	sticker_cooldown = 0 SECONDS
+	///Flag for enabling or disabling trait patches
+	var/patch_traits = FALSE
+	///What type of material we're applying
+	var/datum/xenoartifact_material/material = /datum/xenoartifact_material
+	///Hack fix for double creation, who cares about a debug tool
+	var/skip_label = FALSE
+
+//Create an artifact with all the traits we have selected, but from the item we target
+/obj/item/xenoarchaeology_labeler/debug/afterattack(atom/target, mob/user)
+	target.AddComponent(/datum/component/xenoartifact, material, length(label_traits) ? label_traits : null, TRUE, FALSE, patch_traits)
+	skip_label = TRUE
+	return ..()
+
+/obj/item/xenoarchaeology_labeler/debug/AltClick(mob/user)
+	. = ..()
+	var/choice = tgui_alert(user, "Select Action", "Select Action", list("Toggle Patch", "Change Material"))
+	switch(choice)
+		if("Toggle Patch")
+			patch_traits = !patch_traits
+			to_chat(user, "<span class='notice'>Toggled patch: [patch_traits ? "On" : "Off"].</span>")
+		if("Change Material")
+			var/list/possible_materials = subtypesof(/datum/xenoartifact_material)
+			possible_materials += /datum/xenoartifact_material
```

Comment:
`subtypesof` means that you won't get `/datum/xenoartifact_material` which is defined to be a debug material as parent. Why don't you just use `typesof` when you add the parent manually?